### PR TITLE
Bs dynamic log in

### PR DESCRIPTION
**Pull request by bensenescu**

### CODE REVIEW

File: src/components/AccessControl/AccessControl.vue [hunk 1]
```diff
@@ -18,23 +19,40 @@ export default {
...
+    }),
     userRole() {
       switch (this.userRoleNum) {
         case 0:
           return 'user';
         case 2:
           return 'admin';
         default:
-          return 'error';
+          return 'guest';
```
Comment (jackblanc) [inline src/components/AccessControl/AccessControl.vue:45]: Confused what you're doing here, what about "participating families"?

File: src/store/modules/user.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import userState from '../../utils/auth/state/userState';
```
Comment (jackblanc) [inline src/store/modules/user.js:1]: This file is duplicating functionality @nicoburniske made /utils/auth/state/userState

File: src/components/TheNavigation.vue [hunk 3]
```diff
@@ -6,23 +6,36 @@
     <div class="navlinks">
       <router-link to="/about-us" tag="button">About Us</router-link>
       <router-link to="/events" tag="button">Events</router-link>
-      <router-link to="/login" tag="button">Log In</router-link>
-      <button v-on:click="logout">Log Out</button>
+      <access-control :roles="['guest']">
```
Comment (jackblanc) [inline src/components/TheNavigation.vue:9]: Okay, so this all makes sense. But we should try and keep user concepts and enums consistent across the frontend and the backend. Here's what the backend uses consistently (how consistently @liammoyn ?)

(note: GP and PF should be expanded to longer names that are recognizable. If it doesn't read like english it doesn't work) 

```
public enum PrivilegeLevel {
  GP(0), PF(1), ADMIN(2);

  private int val;

  PrivilegeLevel(int val) {
    this.val = val;
  }

  public int getVal() {
    return val;
  }

  public static PrivilegeLevel from(Integer val) {
    for (PrivilegeLevel privilegeLevel : PrivilegeLevel.values()) {
      if (privilegeLevel.val == val) {
        return privilegeLevel;
      }
    }
    throw new IllegalArgumentException(String.format("Given num (%d) that doesn't correspond to any PrivilegeLevel", val));
  }
}
```

File: src/components/AccessControl/AccessControl.vue (hunk 1 above)
Comment (bensenescu) [reply src/components/AccessControl/AccessControl.vue:45]: I didn't have an account that had pf, that needs to be added as case 1: return 'pf'

File: src/components/TheNavigation.vue (hunk 3 above)
Comment (bensenescu) [reply src/components/TheNavigation.vue:9]: dont have brain capacity to read this, slack me if you need input

**Pull request merged**